### PR TITLE
add keyboard type to search bar

### PR DESCRIPTION
**Pull request by ghuang24**
### Platforms Impacted
- [x] iOS
- [ ] visionOS
- [ ] macOS

### Description of changes

Adding `keyboardType` property to `SearchBar`.

### Binary change

Total increase: 3,584 bytes
Total decrease: 0 bytes
| File | Before | After | Delta |
|------|-------:|------:|------:|
| Total | 31,315,136 bytes | 31,318,720 bytes | ⚠️ 3,584 bytes |
<details>
<summary> Full breakdown </summary>

| File | Before | After | Delta |
|------|-------:|------:|------:|
| SearchBar.o | 381,656 bytes | 384,272 bytes | ⚠️ 2,616 bytes |
| __.SYMDEF | 4,890,864 bytes | 4,891,832 bytes | ⚠️ 968 bytes |
</details>

### Verification

(how the change was tested, including both manual and automated tests)

<details>
<summary>Visual Verification</summary>

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| Simulator Screenshot - iPhone 15 Pro Max - 2024-03-27 at 18 20 02 | Simulator Screenshot - iPhone 15 Pro Max - 2024-03-27 at 18 19 22 |
</details>

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] iOS supported versions (all major versions greater than or equal current target deployment version)
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)
- [ ] iPad [Pointer interaction](https://developer.apple.com/documentation/uikit/pointer_interactions)
- [ ] [SwiftUI](https://developer.apple.com/tutorials/swiftui) consumption (validation or new demo scenarios needed)
- [ ] Objective-C exposure (provide it only if needed)
 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/fluentui-apple/pull/1990)

### CODE REVIEW

**Comment by ghuang24** [comment]
@microsoft-github-policy-service agree company="Microsoft"

File: MicrosoftFluentUI.podspec [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Pod::Spec.new do |s|
   s.name             = 'MicrosoftFluentUI'
-  s.version          = '0.26.0'
+  s.version          = '0.27.0'
```
Comment (sophialee0416) [inline MicrosoftFluentUI.podspec:3]: Version bumps + framework changes should be done in separate PRs

Comment (ghuang24) [reply MicrosoftFluentUI.podspec]: updated

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/SearchBarDemoController.swift [hunk 2]
```diff
@@ -78,8 +78,10 @@ class SearchBarDemoController: DemoController, SearchBarDelegate {
 
         let searchBarNoAutocorrect = buildSearchBar(autocorrectionType: .no, placeholderText: "no autocorrect")
         let searchBarAutocorrect = buildSearchBar(autocorrectionType: .yes, placeholderText: "autocorrect")
+        let numberSearchBar = buildSearchBar(autocorrectionType: .no, placeholderText: "numberpad search")
```
Comment (joannaquu) [inline ios/FluentUI.Demo/FluentUI.Demo/Demos/SearchBarDemoController.swift:81]: nit: can we capitalize n

**Review by joannaquu — APPROVED** [review]

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/SearchBarDemoController.swift (hunk 2 above)
Comment (ghuang24) [reply ios/FluentUI.Demo/FluentUI.Demo/Demos/SearchBarDemoController.swift]: Updated - also capitalized other placeholder texts
